app/services/idv/steps/document_capture_step.rb

```diff
@@ -25,11 +25,33 @@ def extra_view_variables
             image_type: 'selfie',
             transaction_id: flow_session[:document_capture_session_uuid],
           ),
-        }
+        }.merge(native_camera_ab_testing_variables)
       end
 
       private
 
+      def native_camera_ab_testing_variables
+        skip_sdk = randomize?(
+          IdentityConfig.store.idv_native_camera_a_b_testing_percent,
+          flow_session[:document_capture_session_uuid],
+        )
+
+        {
+          native_camera_a_b_testing_enabled:
+            IdentityConfig.store.idv_native_camera_a_b_testing_enabled,
+          native_camera_only: skip_sdk,
+        }
+      end
+
+      def randomize?(target_percent, discriminator)
+        return false if discriminator.blank?
+
+        max_sha = (16 ** 64) - 1
+        user_value = Digest::SHA256.hexdigest(discriminator).to_i(16).to_f / max_sha * 100
+
+        user_value < target_percent.clamp(0, 100)
+      end
```

Comment:
this is copy-pasted wholesale from DocAuthRouter, if we expect to run more A-B tests in the future I think it's time to extract it into a class, give it some clearer names, and get ready to extend it to A-B-C-....N tests

I imagine something like:

```
AbTestBucket.new(buckets: [
  { use_native_capture: 0.5 }, # pull this from config
  { some_other_option: 0.1 }, # example
], default_bucket: :default).get_bucket(uuid)
# => :default
# => :use_native_capture
# etc
```

so then the calling code always has `if bucket == :use_native_capture` or something and it's easy to tell which way a test goes

Reply:
and since 2-case tests are probably the most common, maybe we make two constructors:

```ruby
AbTestBucket.a_b_test(default_bucket: :default, experiment_bucket: :use_native_capture, experiment_percent: 0.5)

AbTestBucket.multibucket_test(
  default_bucket: :default,
  buckets: [
    { use_native_capture: 0.5 },
    { some_other_option: 0.1 },
  ]
)
```

(and we could probably omit `default_bucket: :default` but I'm adding it for clarity)